src/nvim/window.c

```diff
@@ -2453,9 +2453,8 @@ int win_close(win_T *win, bool free_buf)
   }
   if ((firstwin == win && lastwin_nofloating() == win)
       && lastwin->w_floating) {
-    // TODO(bfredl): we might close the float also instead
-    EMSG(e_floatonly);
-    return FAIL;
+    win_close(lastwin, true);
```

Comment:
There is no guarantee that this is the only floating window. Also what happens when `lastwin` is not anchored to `win` (should still be an error)?